### PR TITLE
Dev update auth

### PR DESCRIPTION
**Pull request by Jenjamin3000**
The authentication process has been remodeled and we can now change our password!

### CODE REVIEW

File: app/src/main/java/com/github/campus_capture/bootcamp/fragments/RegisterFragment.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                .setMessage(TOS.TEXT)
+                .show();
+    }
+
+    /**
+     * Sets the Listener on the "already_registered_button"
+     */
+    private void setAlreadyRegisteredButtonListener() {
+        already_registered_button.setOnClickListener(view -> goToSignInFragment("", ""));
+    }
```
Comment (Blopgrop) [inline app/src/main/java/com/github/campus_capture/bootcamp/fragments/RegisterFragment.java:166]: Same here, kinda overcomplicated

File: app/src/main/java/com/github/campus_capture/bootcamp/fragments/RegisterFragment.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        setAlreadyRegisteredButtonListener();
+        setRegisterButtonListener();
+        setSpectatorButtonListener();
+        // Inflate the layout for this fragment
+        return view;
+    }
+
+    private void setRegisterButtonListener(){
+        register_button.setOnClickListener(view -> registerClicked());
+    }
```
Comment (Blopgrop) [inline app/src/main/java/com/github/campus_capture/bootcamp/fragments/RegisterFragment.java:88]: don't need a function : (at 79) `register_button.setOnClickListner(this::registerClicked)`

**Review by Blopgrop — COMMENTED** [review]
Very good work, it seams to work fine and the addition of the fragment navigation gives a more maintainable structure for this part of the app. One note though, could we have a systematic way to navigate through them instead of having to create the same function in each fragment ?

**Comment by Jenjamin3000** [comment]
You are right! I will try to make some changes in order to avoid some duplicated code.

File: app/src/main/java/com/github/campus_capture/bootcamp/fragments/RegisterFragment.java (hunk 2 above)
Comment (Jenjamin3000) [reply app/src/main/java/com/github/campus_capture/bootcamp/fragments/RegisterFragment.java:88]: But I think it modularizes the code and so makes it easier to read. Isn't it?

File: app/src/main/java/com/github/campus_capture/bootcamp/activities/AuthenticationActivity.java
Comment (Blopgrop) [inline app/src/main/java/com/github/campus_capture/bootcamp/activities/AuthenticationActivity.java:1]: yeah that's an ez fix

**Review by Blopgrop — APPROVED** [review]
Yeah I think that's better like that, good job again

**Pull request merged**